packages/connector-thegraph/src/index.ts

```diff
@@ -21,48 +37,52 @@ class ConnectorTheGraph implements ConnectorInterface {
     // this.#appClient = createClient({ url: appSubgraphUrl('app_id') })
   }
 
-  async permissions(orgAddress: string): Promise<Permission[]> {
-    const query = `
-      query {
-        organization(id: "${orgAddress}") {
-          acl {
-            permissions {
-              entity
-              app {
-                address
-              }
-              role {
-                name
-                manager
-              }
-            }
-          }
-        }
-      }
-    `
-
-    const res = await this.#daoClient.query(query).toPromise()
-
-    const { permissions } = res?.data?.organization.acl
-    if (!permissions) {
-      return []
-    }
-
-    return permissions.map(
-      ({
-        app,
-        entity,
-        role,
-      }: {
-        app: any
-        entity: string
-        role: { name: string }
-      }) => ({
-        app: app?.address || '',
-        entity,
-        role: role.name,
-      })
-    )
+  async permissionsForOrg(orgAddress: string): Promise<Permission[]> {
+    const org = (await this._performQuery(
+      queries.ORGANIZATION_PERMISSIONS,
+      { orgAddress }
+    )).organization as OrganizationDataGql
+
+    return parsePermissions(this, org.acl?.permissions)
```

Comment:
Wondering if we should use optional chaining at every level to be safe, every time an object is built from an external source?

```suggestion
    const org = (await this._performQuery(
      queries.ORGANIZATION_PERMISSIONS,
      { orgAddress }
    ))?.organization as OrganizationDataGql

    return parsePermissions(this, org?.acl?.permissions)
```